packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2127,63 +2117,137 @@ class TextSelectionGestureDetectorBuilder {
   ///  * [TextSelectionGestureDetector.onDragSelectionUpdate], which triggers
   ///    this callback./lib/src/material/text_field.dart
   @protected
-  void onDragSelectionUpdate(DragStartDetails startDetails, DragUpdateDetails updateDetails) {
+  void onDragSelectionUpdate(DragUpdateDetails details, TapStatus status) {
     if (!delegate.selectionEnabled) {
       return;
     }
 
-    if (!_isShiftTapping) {
+    if (!status.isShiftPressed) {
       // Adjust the drag start offset for possible viewport offset changes.
       final Offset startOffset = renderEditable.maxLines == 1
           ? Offset(renderEditable.offset.pixels - _dragStartViewportOffset, 0.0)
           : Offset(0.0, renderEditable.offset.pixels - _dragStartViewportOffset);
+      final Offset dragStartGlobalPosition = details.globalPosition - details.offsetFromOrigin;
```

Comment:
Is this something that you could always do to get the start position?  It seems nice that we don't need DragStartDetails any more.

Reply:
Yes! because I added `offsetFromOrigin` to `DragUpdateDetails`. This is similar to `LongPressMoveUpdateDetails` which also has `offsetFromOrigin`. A good use case where that is currently used is in TextField https://github.com/flutter/flutter/blob/e3c29ec00c9c825c891d75054c63fcc46454dca1/packages/flutter/lib/src/material/text_field.dart#L76 .